namer/consul/src/main/scala/io/buoyant/namer/consul/SvcCache.scala

```diff
@@ -0,0 +1,98 @@
+package io.buoyant.namer.consul
+
+import com.twitter.finagle._
+import com.twitter.util._
+import io.buoyant.consul.v1
+import io.buoyant.namer.Metadata
+
+private[consul] case class SvcKey(name: String, tag: Option[String]) {
+  override def toString = tag match {
+    case Some(t) => s"$name:$t"
+    case None => name
+  }
+}
+
+/**
+ * Contains all cached serviceNodes responses for a particular serviceName
+ * in a particular datacenter
+ */
+private[consul] class SvcCache(
+  consulApi: v1.ConsulApi,
+  datacenter: String,
+  key: SvcKey,
+  domain: Option[String]
+) {
+
+  private[this] val updatableAddr = Var[Addr](Addr.Pending)
+  def addrs: VarUp[Addr] = updatableAddr
+
+  @volatile private[this] var index = "0"
+  private[this] def setIndex(idx: Option[String]) = idx match {
+    case None =>
+    case Some(idx) =>
+      index = idx
+  }
+
+  private[this] val indexed: v1.Indexed[Seq[v1.ServiceNode]] => Seq[v1.ServiceNode] = {
+    case v1.Indexed(nodes, idx) =>
+      setIndex(idx)
+      nodes
+  }
+
+  private[this] def mkRequest(): Future[Seq[v1.ServiceNode]] =
+    consulApi.serviceNodes(
+      key.name,
+      datacenter = Some(datacenter),
+      tag = key.tag,
+      blockingIndex = Some(index),
+      retry = true
+    ).map(indexed)
+
+  def clear(): Unit = {
+    updatableAddr() = Addr.Pending
+  }
+
+  private[this] def run(): Future[Unit] =
+    mkRequest().flatMap(update).handle(handleUnexpected)
+
+  def update(nodes: Seq[v1.ServiceNode]): Future[Unit] = {
+    val addr = nodes.flatMap(serviceNodeToAddr) match {
+      case addrs if addrs.isEmpty => Addr.Neg
+      case addrs =>
+        val meta = domain match {
+          case None => Addr.Metadata.empty
+          case Some(domain) =>
```

Comment:
This is common and intentional throughout Finagle (and while we don't use Finagle's style as dogma, I find it's convenient for the projects to stay aligned).

When we have box types like Option, it's preferred to simply use a name like `domain` rather than `domainOpt` or `maybeDomain` etc.  Because we're just unboxing the value, it's fine to shadow the name because it signals the intent of "just unboxing this thing".  I suppose it'd also be fine to use a temp val like `d` here.